plugins/topology/src/utils/pod-resource-utils.ts

```diff
@@ -252,6 +290,75 @@ const getPodsForDeployment = (
   };
 };
 
+export const getPodsForStatefulSet = (
+  statefulSet: V1StatefulSet,
+  resources: K8sResponseData,
+): PodRCData => {
+  const obj = {
+    ...statefulSet,
+  };
+  const statefulSets = getStatefulSetsResource(obj, resources);
+  const [current, previous] = statefulSets;
+  const isRollingOut = !!current && !!previous;
+
+  return {
+    obj,
+    current: current as PodControllerOverviewItem,
+    previous: previous as PodControllerOverviewItem,
+    isRollingOut,
+    pods: [...(current?.pods || []), ...(previous?.pods || [])] as V1Pod[],
+  };
+};
+
+export const getJobsForCronJob = (
+  cronJobUid: string,
+  resources: K8sResponseData,
+) => {
+  if (!resources?.jobs?.data?.length || resources?.jobs?.data?.length === 0) {
+    return [];
+  }
+  return resources.jobs.data.filter(job =>
+    job.metadata?.ownerReferences?.find(ref => ref.uid === cronJobUid),
+  );
+};
+
+export const getPodsForCronJob = (
+  cronJob: V1CronJob,
+  resources: K8sResponseData,
+): PodRCData => {
+  const obj = {
+    ...cronJob,
+  };
+  const jobs = getJobsForCronJob(obj?.metadata?.uid as string, resources);
+  return {
+    obj,
+    current: undefined,
+    previous: undefined,
+    isRollingOut: undefined,
+    pods: jobs?.reduce((acc: V1Pod[], job) => {
+      acc.push(...getPodsForResource(job, resources));
+      return acc;
+    }, []),
+  };
+};
+
+export const getPodsForDaemonSet = (
+  daemonSet: V1DaemonSet,
+  resources: K8sResponseData,
+): PodRCData => {
+  const obj = {
+    ...daemonSet,
+    kind: DaemonSetGVK.kind,
+  };
```

Comment:
do check once if `daemonSet` has kind we don't need to get it from `DaemonSetGVK.kind`